packages/playwright-core/src/server/trace/recorder/tracing.ts

```diff
@@ -171,6 +172,7 @@ export class Tracing extends SdkObject implements InstrumentationListener, Snaps
       this._startScreencast();
     if (state.options.snapshots)
       await this._snapshotter?.start();
+    return { traceName: state.traceName };
```

Comment:
`_changeTraceName` does not update `state.traceName` synchronously, so this might be a wrong one.